File: Revised/Py/Mat/MatReducer.py
def multiply_matrices(a, b):
    m = len(a)
    n = len(b[0])
    p = len(b)
    result = [[0] * n for _ in range(m)]

    for i in range(m):
        for j in range(n):
            for k in range(p):
                result[i][j] += a[i][k] * b[k][j]

    return result

def print_matrix(matrix):
    for row in matrix:
        print " ".join(map(str, row))

if __name__ == "__main__":
    matrix1 = [[0] * 3 for _ in range(2)]
    matrix2 = [[0] * 2 for _ in range(3)]

    try:
        with open("output.txt", "r") as reader:
            for line in reader:
                parts = line.strip().split(",")
                if parts[0] == "M":
                    i, j, value = map(int, parts[1:])
                    matrix1[i][j] = value
                elif parts[0] == "N":
                    i, j, value = map(int, parts[1:])
                    matrix2[i][j] = value

        result = multiply_matrices(matrix1, matrix2)
        print_matrix(result)
    except IOError as e:
        print "Error reading from output.txt:", e

File: Revised/Jav/Mat/MatReducer.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class MatReducer {
    public static void main(String[] args) {
        int[][] matrix1 = new int[2][3];
        int[][] matrix2 = new int[3][2];

        try {
            BufferedReader reader = new BufferedReader(new FileReader("output.txt"));
            String line;
            while ((line = reader.readLine()) != null) {
                String[] parts = line.split(",");
                if (parts[0].equals("M")) {
                    int i = Integer.parseInt(parts[1]);
                    int j = Integer.parseInt(parts[2]);
                    int value = Integer.parseInt(parts[3]);
                    matrix1[i][j] = value;
                } else if (parts[0].equals("N")) {
                    int i = Integer.parseInt(parts[1]);
                    int j = Integer.parseInt(parts[2]);
                    int value = Integer.parseInt(parts[3]);
                    matrix2[i][j] = value;
                }
            }
            reader.close();

            int[][] result = multiplyMatrices(matrix1, matrix2);
            printMatrix(result);
        } catch (IOException e) {
            System.err.println("Error reading from output.txt: " + e.getMessage());
        }
    }

    private static int[][] multiplyMatrices(int[][] a, int[][] b) {
        int m = a.length;
        int n = b[0].length;
        int p = b.length;
        int[][] result = new int[m][n];

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < p; k++) {
                    result[i][j] += a[i][k] * b[k][j];
                }
            }
        }

        return result;
    }

    private static void printMatrix(int[][] matrix) {
        for (int[] row : matrix) {
            for (int value : row) {
                System.out.print(value + " ");
            }
            System.out.println();
        }
    }
}

